src/crud_internal.c

```diff
@@ -448,208 +438,202 @@ int retrieveObject( wrp_msg_t *reqMsg, wrp_msg_t **response )
 	char *str1 = NULL;
 	const char *parse_error = NULL;
 
-	cJSON *jsonresponse = cJSON_CreateObject();
+	if(reqMsg->u.crud.dest !=NULL)
+	{
+		ParodusInfo("reqMsg->u.crud.dest is %s\n", reqMsg->u.crud.dest);
```

Comment:
The block that breaks the destination into the components should leverage: `wrp_get_msg_dest_element()`.

Reply:
Incorporated. Used wrp_get_msg_dest_element() and also added single function parse_dest_elements_to_string() to get pre-separated strings for each crud operation.

wrp_get_msg_dest_element() method from wrp-c is modified to support next element as crud requires dest format as mac:44aaf59b184a/parodus/tags/test.
Submitted PR for this change.
https://github.com/Comcast/wrp-c/pull/62